server/api/org.go

```diff
@@ -15,41 +15,121 @@
 package api
 
 import (
+	"errors"
 	"net/http"
+	"strconv"
+	"strings"
 
 	"github.com/woodpecker-ci/woodpecker/server"
 	"github.com/woodpecker-ci/woodpecker/server/model"
 	"github.com/woodpecker-ci/woodpecker/server/router/middleware/session"
+	"github.com/woodpecker-ci/woodpecker/server/store"
+	"github.com/woodpecker-ci/woodpecker/server/store/types"
 
 	"github.com/gin-gonic/gin"
 )
 
+// GetOrg
+//
+//	@Summary	Get organization by id
+//	@Router		/orgs/{org_id} [get]
+//	@Produce	json
+//	@Success	200	{array}	Org
+//	@Tags		Organization
+//	@Param		Authorization	header	string	true	"Insert your personal access token"	default(Bearer <personal access token>)
+//	@Param		org_id			path	string	true	"the organziation's id"
+func GetOrg(c *gin.Context) {
+	_store := store.FromContext(c)
+
+	orgID, err := strconv.ParseInt(c.Param("org_id"), 10, 64)
+	if err != nil {
+		c.String(http.StatusBadRequest, "Error parsing org id. %s", err)
+		return
+	}
+
+	org, err := _store.OrgFind(orgID)
+	if err != nil {
+		if errors.Is(err, types.RecordNotExist) {
+			c.AbortWithStatus(http.StatusNotFound)
+			return
+		}
+		_ = c.AbortWithError(http.StatusInternalServerError, err)
+		return
+	}
+
+	c.JSON(http.StatusOK, org)
+}
+
 // GetOrgPermissions
 //
 //	@Summary	Get the permissions of the current user in the given organization
-//	@Router		/orgs/{owner}/permissions [get]
+//	@Router		/orgs/{org_id}/permissions [get]
 //	@Produce	json
 //	@Success	200	{array}	OrgPerm
 //	@Tags		Organization permissions
 //	@Param		Authorization	header	string	true	"Insert your personal access token"	default(Bearer <personal access token>)
-//	@Param		owner			path	string	true	"the owner's name"
+//	@Param		org_id			path	string	true	"the organziation's id"
 func GetOrgPermissions(c *gin.Context) {
-	var (
-		err   error
-		user  = session.User(c)
-		owner = c.Param("owner")
-	)
+	user := session.User(c)
+	_store := store.FromContext(c)
+
+	orgID, err := strconv.ParseInt(c.Param("org_id"), 10, 64)
+	if err != nil {
+		c.String(http.StatusBadRequest, "Error parsing org id. %s", err)
+		return
+	}
 
 	if user == nil {
 		c.JSON(http.StatusOK, &model.OrgPerm{})
 		return
 	}
 
-	perm, err := server.Config.Services.Membership.Get(c, user, owner)
+	org, err := _store.OrgFind(orgID)
 	if err != nil {
-		c.String(http.StatusInternalServerError, "Error getting membership for %q. %s", owner, err)
+		c.String(http.StatusInternalServerError, "Error getting org %d. %s", orgID, err)
+		return
+	}
+
+	if (org.IsUser && org.Name == user.Login) || user.Admin {
```

Comment:
we should handle the case here where the org is actually a user but migration got it wrong